src/compile/selection/transforms/project.ts

```diff
@@ -139,15 +139,20 @@ const project: TransformCompiler = {
       if (scales.has(selCmpt)) {
         log.warn(log.message.NO_INIT_SCALE_BINDINGS);
       } else {
-        const parseInit = <T extends SelectionInitMapping | SelectionInitIntervalMapping>(i: T): T['a'][] => {
+        function parseInit(i: SelectionInitMapping): SelectionInitMapping['a'];
+        function parseInit(i: SelectionInitIntervalMapping): SelectionInitIntervalMapping['a'];
+        // eslint-disable-next-line no-inner-declarations
+        function parseInit(i: SelectionInitMapping | SelectionInitIntervalMapping): any {
```

Comment:
Better remove `SelectionInitIntervalMapping` and make `SelectionInitMapping` a generic with two forms: 

`SelectionInitMapping<Value | DateTime>` and `SelectionInitMapping<SelectionInitInterval>`

Reply:
Great idea. `SelectionInitMapping` == `Dict` then.